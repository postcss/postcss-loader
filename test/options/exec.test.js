const { webpack } = require('@webpack-utilities/test')

describe('Options', () => {
  test('Exec - {Boolean}', () => {
    const config = {
      loader: {
        test: /style\.(exec\.js|js)$/,
        options: {
          exec: true
        }
      }
    }

    return webpack('jss/exec/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1]

      expect(source).toEqual(
        'module.exports = "a {\\n    color: green\\n}"'
      )

      expect(source).toMatchSnapshot()
    })
  })

  test('JSS - {String}', () => {
    const config = {
      loader: {
        options: {
          parser: 'postcss-js'
        }
      }
    }

    return webpack('jss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1]

      expect(source).toMatchSnapshot()
    })
  })
})
