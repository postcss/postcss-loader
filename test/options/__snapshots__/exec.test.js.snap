// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Exec should work Exec - {Boolean}: css 1`] = `
"a {
    color: green
}"
`;

exports[`Options Exec should work Exec - {Boolean}: errors 1`] = `Array []`;

exports[`Options Exec should work Exec - {Boolean}: warnings 1`] = `Array []`;

exports[`Options Exec should work JSS - {String}: css 1`] = `
"{ a: { color: 'yellow' } }
"
`;

exports[`Options Exec should work JSS - {String}: errors 1`] = `Array []`;

exports[`Options Exec should work JSS - {String}: warnings 1`] = `Array []`;
